server/room.ts

```diff
@@ -650,6 +660,24 @@ export class Room {
     this.addChatMessage(socket, chatMsg);
   };
 
+  private toggleChat = async (
+    socket: Socket,
+    data: { uid: string; token: string; toggleChat: boolean }
+  ) => {
+    if (!data) {
+      return;
+    }
+    const decoded = await validateUserToken(data.uid, data.token);
+    if (!decoded) {
+      return;
+    }
+    if (!this.validateLock(socket.id)) {
+      return;
+    }
+    this.isChatEnabled = data.toggleChat;
```

Comment:
if we don't serialize this to db, then it'll default to off every server reboot? probably need to make it a serialized field